hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -377,9 +388,21 @@ public void write(HoodieRecord record, Option<IndexedRecord> insertValue) {
         writer.close();
 
         // update final size, once for all log files
+        long totalLogFileSize = 0;
         for (WriteStatus status: statuses) {
           long logFileSize = FSUtils.getFileSize(fs, new Path(config.getBasePath(), status.getStat().getPath()));
           status.getStat().setFileSizeInBytes(logFileSize);
+          totalLogFileSize += logFileSize;
+        }
+
+        if (config.isMetricsOn() && config.shouldCollectObservabilityMetrics()) {
```

Comment:
This code is repeated across handles. So probably move it within the table itself and simplify the check for metrics enabled.

hoodieTable.updateObservabilityMetrics(xxx)

or 

hoodieTable.getObservabilityMetrics().ifPresent(....)

Reply:
Done.